KIPs/kip-163.md

```diff
@@ -0,0 +1,531 @@
+
+#### IPublicDelegation
+
+IPublicDelegation must be implemented in `PublicDelegation` to be fully compatible with CnStakingV3:
+
+##### Interface
+
+```solidity
+pragma solidity 0.8.25;
+
+/// @title KIP-163 Public Delegation Interface
+/// @dev See https://github.com/klaytn/kips/issues/163
+interface IPublicDelegation {
+    /// @dev Stake KLAY for the _recipient.
+    /// It is used in CnStakingV3.handleRedelegation to stake KLAY for the _recipient when handling redelegation.
+    /// It must stake KLAY to the CnStakingV3 in the same transaction.
+    /// See CnStakingV3.handleRedelegation for more details.
+    /// @param _recipient The address to stake for
+    function stakeFor(address _recipient) external payable;
+
+    /// @dev Returns the current rewards to be automatically compounded during the `stakeFor` function.
+    /// It is used in CnStakingV3.handleRedelegation to calculate the expected balance after calling `stakeFor`.
+    /// If implemented public delegation doesn't support auto-compounding during the `stakeFor`, it should return 0.
+    /// See CnStakingV3.handleRedelegation for more details.
+    function reward() external view returns (uint256);
+}
+```
+
+Also, all staking operations in CNv3 must be done through public delegation, so there is no fixed interface, but it should call CNv3 appropriately.
+
+##### Methods
+
+```solidity
+function stakeFor(address _recipient) payable;
+```
+
+It is used in `CnStakingV3.handleRedelegation` to stake KLAY for the `_recipient` when handling redelegation. The `stakeFor` MUST not revert `handleRedelegation`.
+
+```solidity
+function reward() view returns (uint256);
+```
+
+It returns the current rewards to be automatically compounded during the `stakeFor` function. It is used in `CnStakingV3.handleRedelegation` to calculate the expected balance after calling `stakeFor` function. If the public delegation doesn’t support auto-compounding during the `stakeFor`, it must return 0, otherwise exact rewards to be compounded. The `reward` MUST not revert `handleRedelegation`.
+
+#### PublicDelegation
+
+PD is a public delegation contract based on ERC4626. It must be set up through `setPublicDelegation` in CNv3. Since PD must receive and calculate block rewards, the reward address of the CNv3 must be set to PD. If validator deploys multiple CNv3s to use both initial lockup and PD, the reward address for all CNv3s must be PD (i.e., the reward for initial lockup is also sent to PD).
+
+When block reward is auto-compounding, the commission is calculated and paid automatically.
+
+Since PD is a type of token contract, its name and symbol will be determined as follows:
+
+-   Name: `{gcName} Public Delegated KLAY` (ex. `KF Public Delegated KLAY`)
+-   Symbol: `{gcName}-pdKLAY` (ex. `KF-pdKLAY`)
+
+##### Interface
+
+```solidity
+pragma solidity 0.8.25;
+
+import "../CnV3/ICnStakingV3.sol";
+import "./IPublicDelegation.sol";
+
+interface IPublicStakedKlay is IPublicDelegation {
```

Comment:
Can we use 'IKIP163' instead of `IPublicDelegation`, and `IPublicDelegation` instead of `IPublicStakedKlay`?

Reply:
Oh, my bad. Use `IPublicDelegation` instead of `IPublicStakedKlay`.

Applied at https://github.com/klaytn/kips/pull/164/commits/1932c1bb534b2b9f49f9a69fa47daf468549e3ee